modules/ocf_irc/templates/inspircd.conf.erb

```diff
@@ -0,0 +1,202 @@
+# Port 25580 is localhost-only for anope services to connect
+<bind address="" port="6661-6669" type="clients">
+<bind address="" port="6697" type="clients" ssl="gnutls">
+<bind address="127.0.0.1" port="25580" type="servers">
+
+# Link to the anope server, only allow local connections
+<link name="services.irc.ocf.berkeley.edu"
+      ipaddr="localhost"
+      port="25580"
+      allowmask="127.0.0.0/8"
+      sendpass="<%= @passwords['anope-link'] -%>"
+      recvpass="<%= @passwords['anope-link'] -%>">
+
+<uline server="services.irc.ocf.berkeley.edu" silent="yes">
+
+
+# Passwords needed to kill/restart the IRCd while connected to it
+<power diepass="<%= @passwords['diepass'] -%>"
+       restartpass="<%= @passwords['restartpass'] -%>"
+       pause="5">
+
+
+# Users connecting from inside the OCF subnet have less restrictions and have
+# their host cloaked (+x mode)
+<connect allow="169.229.226.0/24"
+         timeout="60" flood="20" localmax="1000" globalmax="1000"
+         limit="1000" modes="+x">
+
+<connect allow="2607:f140:8801::/64"
+         timeout="60" flood="20" localmax="1000" globalmax="1000"
+         limit="1000" modes="+x">
+
+<connect allow="*"
+         timeout="60" flood="20" threshold="20" fakelag="on"
+         pingfreq="120" sendq="262144" recvq="8192"
+         useident="no" localmax="5" globalmax="5">
+
+# Cloaking for the OCF subnet
+<cloak mode="full" key="<%= @passwords['cloak-key'] -%>" prefix="OCF-">
+
+
+# Allow overrides when users are opers
+<override noisy="yes" requirekey="no">
+
+# Create different classes for allowing access to groups of commands
+<class name="Shutdown"
+       commands="DIE RESTART REHASH LOADMODULE UNLOADMODULE RELOAD">
+<class name="ServerLink"
+       commands="CONNECT SQUIT RCONNECT MKPASSWD MKSHA256">
+<class name="BanControl"
+       commands="KILL GLINE KLINE ZLINE QLINE ELINE">
+<class name="OperChat"
+       commands="WALLOPS GLOBOPS SETIDLE SPYLIST SPYNAMES">
+<class name="HostCloak"
+       commands="SETHOST SETIDENT SETNAME CHGHOST CHGIDENT">
+
+# NetAdmins can pretty much do everything
+<type name="NetAdmin"
+      classes="OperChat BanControl HostCloak Shutdown ServerLink"
+      vhost="ocf.gods"
+      override="KICK MODEOP MODEDEOP MODEVOICE MODEDEVOICE MODEHALFOP MODEDEHALFOP OTHERMODE">
+
+# Allow oper authentication using LDAP
+<ldapoper baserdn="ou=People,dc=OCF,dc=Berkeley,dc=EDU"
+          server="ldaps://ldap.ocf.berkeley.edu"
+          searchscope="subtree"
+          attribute="uid">
+
+# List of allowed opers
+<oper name="waf"          password="uid=waf"      host="*@*" type="NetAdmin" hash="ldap">
+<oper name="nattofriends" password="uid=tzhu"     host="*@*" type="NetAdmin" hash="ldap">
+<oper name="ckuehl"       password="uid=ckuehl"   host="*@*" type="NetAdmin" hash="ldap">
+<oper name="jvperrin"     password="uid=jvperrin" host="*@*" type="NetAdmin" hash="ldap">
+<oper name="mattmcal"     password="uid=mattmcal" host="*@*" type="NetAdmin" hash="ldap">
```

Comment:
Ok, I changed this so that it stores oper credentials as sha256 hashed passwords in mysql. Stronger encryption would be better, like [using bcrypt or something](https://github.com/inspircd/inspircd/issues/741), but the strongest option available is sha256. Unfortunately, any opers will have to go in to mysql and add their own hashed password, but at least it's not the same password as for admin/root access, which is definitely better.